scripts/apps/authoring/track-changes/inline-comments.js

```diff
@@ -86,6 +111,48 @@ function InlineCommentsCtrl($scope, userList, metadata, content) {
             return;
         }
 
+        unresolvedComments = unresolvedComments.map(({fieldId, comments, ...rest}) => {
```

Comment:
I see you didn't take my advice to check out how `generateAnnotations` is implemented and decided to write it from scratch :grin: `generateAnnotations` calls `getCustomMetadata` which already has all the crawling the `ContentState` for unique stylenames implemented. Actually you would need to refactor it and extract annotation specific stuff to a different function, but otherwise it's 90% ready. I think it'd be best to drop this block of code and retrieve comments from content using `getCustomMetadata`.

Reply:
I did try to do it but as I explained above it didn't work for me.

Reply:
Don't make any changes for a while now, I'll try to fork your PR tomorrow and see if I can make it work using `getCustomMetadata`

Reply:
I've refactored the code to get comments from text the same way as annotations https://github.com/tomaskikutis/superdesk-client-core/commit/944a5e032c85b8343495ef422686bd7dc7fa0c30

Merge this, and this PR is almost good. One more thing I'd like you to look into is if you can remove `commentedText` from `highlight.data`. It's better to have consistent data structures and not modify them if you need an additional property. It's better to add that additional property outside of a common data structure
```
{
    "highlight": getHighlight(),
    "comment-specific-properties": {
        "commentedText": "abc"
    }
}
```

Reply:
Okay, thanks @tomaskikutis. About the commentedText, I just copied the [structure we have for resolved comments](https://github.com/superdesk/superdesk-client-core/blob/20a5ee27b2c2218ea048f978cb54ac128a125fa3/scripts/core/editor3/components/comments/CommentPopup.jsx#L99). If I change this one I should change both.

Reply:
Let's keep it then. Let's also keep it immutable and not modify the original highlight object. I think I already did that after forking this PR.